File: Frontend/lib/features/profile/domain/entities/patient.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:practice/features/appointments/domain/entities/appointment.dart';
import 'package:practice/features/exercises/domain/entities/exercise_assignment.dart';

part 'patient.freezed.dart';
part 'patient.g.dart';

@freezed
abstract class Patient with _$Patient {
  const factory Patient({
    required String patientId,
    required String firstName,
    required String lastName,
    required String phoneNumber,
    required String email,
    required String diagnosis,
    DateTime? createdAt,
    @Default(false) bool isActive,
    @Default([]) List<ExerciseAssignment> exercises,
    @Default([]) List<Appointment> appointments,
  }) = _Patient;

  factory Patient.fromJson(Map<String, dynamic> json) =>
      _$PatientFromJson(json);
}

File: Backend/PhysioLink.Domain/Entities/Patient.cs


namespace PhysioLink.Domain.Entities{
public class Patient : ClinicScopedEntity
{
    public Guid PatientId { get; set;}
    public string FirstName { get; set; }

    public string LastName { get; set;}
    public string PhoneNumber { get; set;}

    public Guid ApplicationUserId {get; set;}
    public Guid? TherapistId { get; set; }

    public Therapist? Therapist { get; set; }
    public Clinic? Clinic { get; set; }

       public string? Email { get; set;}
    public string Username { get; set; }
    public string Diagnosis { get; set;}

    public bool IsActive { get; set; }

    public Patient(string firstName, string lastName, string phoneNumber, Guid applicationUserId, string username, string? email, string diagnosis)
        {
            PatientId = Guid.NewGuid();
            FirstName = firstName;
            LastName = lastName;
            Username = username;
            Email = email;
            Diagnosis = diagnosis;
            PhoneNumber = phoneNumber;
            ApplicationUserId = applicationUserId;
            IsActive = true;
        }
    
}
}